carmm/analyse/povray_render.py

```diff
@@ -0,0 +1,98 @@
+from ase.io import read, write
+from ase.data.colors import jmol_colors
+from ase.data import atomic_numbers
+import subprocess
+
+
+def povray_render(atoms, output='povray', view=False, atom_subs=None,
+                  generic_projection_settings=None, povray_settings=None):
+    """
+    Saves a .png file showing a povray rendered visualisation of an atoms object.
+
+    Parameters:
+
+    atoms: Atoms object
+        Structure to be visualised
+    output: String
+        Output file basename i.e. 'output'.pov, 'output'.ini etc.
+    view: Boolean
+        Show the output window
+    atom_subs: List of Lists of Strings
+        Pairs of atomic symbols with the first being changed to the second for clearer visualisation
+        Tip: Find a second atom with a similar atomic radius to the first with a more distinctive colour
+    generic_projection_settings: Dictionary
+        Settings used by PlottingVariables
+        (see https://gitlab.com/ase/ase/-/blob/master/ase/io/utils.py PlottingVariables/__init__ for settings options)
+    povray_settings: Dictionary
+        Settings used by Povray for visualisation
+        (see https://gitlab.com/ase/ase/-/blob/master/ase/io/pov.py POVRAY/__init__ for settings options)
+
+    Returns:
+
+    A .png file of the atoms object, visualised in Povray with the desired settings
+    """
+
+    # Default visual settings
+    if generic_projection_settings is None:
+        generic_projection_settings = {}
+
+    if 'rotation' not in generic_projection_settings:
+        generic_projection_settings['rotation'] = '0x,0y,0z'
+    if 'radii' not in generic_projection_settings:
+        generic_projection_settings['radii'] = 1.0
+
+    if 'colors' not in generic_projection_settings:
+        generic_projection_settings['colors'] = None
+    else:
+        for atom in atoms:
+            sym = atom.symbol
+            if sym not in generic_projection_settings['colors']:
+                generic_projection_settings['colors'][sym] = jmol_colors[atomic_numbers[sym]]
```

Comment:
I am just curious. We can set the color of a certain element without replacing it with another atom, right?

Reply:
Yeah, this is the code for doing that. So the choice is either to substitute one element for another with a similar radius but clearer colour (e.g. I substitute Al for Kr a lot), or you can define a dictionary of element colours that you want to change and this code that you highlighted will go through and fill out the rest of the dictionary with their default jmol colours

Reply:
Happy for me to merge the pull request?